samza-core/src/main/java/org/apache/samza/config/TaskConfig.java

```diff
@@ -65,6 +65,20 @@ public class TaskConfig extends MapConfig {
   public static final String COMMIT_TIMEOUT_MS = "task.commit.timeout.ms";
   static final long DEFAULT_COMMIT_TIMEOUT_MS = Duration.ofMinutes(30).toMillis();
 
+  // Flag to indicate whether to skip commit during failures (exceptions or timeouts)
+  public static final String SKIP_COMMIT_DURING_FAILURES_ENABLED = "task.commit.skip.commit.during.failures.enabled";
+  private static final boolean DEFAULT_SKIP_COMMIT_DURING_FAILURES_ENABLED = false;
+
+  // Maximum number of allowed commit exceptions.
+  // If the number of commit exceptions exceeds this limit, the task will be shut down.
+  public static final String SKIP_COMMIT_EXCEPTION_MAX_LIMIT = "task.commit.skip.commit.exception.max.limit";
+  private static final int DEFAULT_SKIP_COMMIT_EXCEPTION_MAX_LIMIT = 5;
+
+  // Maximum number of allowed commit timeouts.
```

Comment:
i think we also need to add the word "successive". allowed successive timeouts. Because if a commit succeeds, we are resetting the counters.